src/common/client_test.go

```diff
@@ -0,0 +1,61 @@
+package common
+
+import (
+	"testing"
+
+	"github.com/opslevel/opslevel-go"
+	"github.com/rocktavious/autopilot"
+)
+
+func TestServiceNeedsUpdate(t *testing.T) {
```

Comment:
1. It might be better to break this up into multiple tests. Having it be 1 big test is hard to read - ex. to tell what the correct output is for each test, you have to scroll down to the assert section and you lose context about what the input was for that test case.
2. The code won't ever update a field to `""`, is that correct? Ex. The k8s sync detects a service's description is empty in k8s, should it get set to empty in OL too?
3. One more test case that would be good to add - input field is `""`, `needsUpdate` should return `false`

Reply:
1. I was trying to reduce code duplication - maybe theres a better way to achieve that.

2. This is true and also a known problem we are going to try and fix in Q2/Q3 - its a root problem with our graphql even if we sent through an `""` it wouldn't update to that because of a lower level problem.

3. Are you suggesting the input to `serviceNeedsUpdate` func to be `""`. If so thats not a valid go code because golang won't let you pass an empty string for an object - the types have to match - thats a limitation of strongly typed languages.

Reply:
1. Fair enough. We normally define a function like `AssertResult` where you can define the common code like calling your function, asserting the result. Then each test calls this function and passes in the specific input and output for that test - ex: `TestAssert({ Name: 'Test'}, false)`. This is great because it balances readability (new people can easily tell what the input / expected output is for each test) with modularization.
2. Fair enough.
3. My bad, I meant testing stuff like `{ Owner: "" }`. Since we have code to handle that, we should test it.

Reply:
I split the tests by IsTrue and IsFalse - i like that alot better

Reply:
1. I like that idea - i'll apply it to the next set of tests I write.

3. These are being implicitly set by golang because golang has the concept of pointers and since the fields are not of pointer types they cannot be nil and thus default to an empty string automatically.